### PR TITLE
antialias by default on desktop too

### PR DESCRIPTION
**Pull request by ngokevin**
**Description:**

Anti-aliasing is desirable for VR. If we're targeting room scale, enabling it makes sense as the default.

### CODE REVIEW

File: src/core/scene/a-scene.js [hunk 1]
```diff
@@ -344,10 +344,10 @@ module.exports = registerElement('a-scene', {
         var canvas = this.canvas;
         // Set at startup. To enable/disable antialias
         // at runttime we would have to recreate the whole context
-        var antialias = this.getAttribute('antialias') === 'true';
+        var antialias = this.getAttribute('antialias') !== 'false';
```
Comment (dmarcos) [inline src/core/scene/a-scene.js]: We can enable antialias by default only on desktop

**Comment by ngokevin** [comment]
Updated, with tests. Logic is (explicitly enabled or disabled) or (has native webvr or is not mobile)

**Comment by ngokevin** [comment]
@dmarcos r?

File: src/core/scene/a-scene.js [hunk 2]
```diff
@@ -518,3 +514,19 @@ function exitFullscreen () {
...
+
+/**
+ * Determines if renderer anti-aliasing should be enabled.
+ * Enabled by default if has native WebVR or is desktop.
+ *
+ * @returns {bool}
+ */
+function shouldAntiAlias (sceneEl) {
+  // Explicitly set.
+  if (sceneEl.getAttribute('antialias') === 'true') { return true; }
```
Comment (dmarcos) [inline src/core/scene/a-scene.js]: Can we do ` if (sceneEl.getAttribute('antialias') !== null) { return sceneEl.getAttribute('antialias') === 'true'; }`

File: src/core/scene/a-scene.js [hunk 3]
```diff
@@ -518,3 +514,19 @@ function exitFullscreen () {
...
+  if (sceneEl.getAttribute('antialias') === 'false') { return false; }
+
+  // Has WebVR or is desktop.
+  return window.hasNativeWebVRImplementation || !sceneEl.isMobile;
```
Comment (cvan) [inline src/core/scene/a-scene.js]: this will match Daydream, FYI

Comment (cvan) [inline src/core/scene/a-scene.js]: this okay?

File: src/core/scene/a-scene.js [hunk 4]
```diff
@@ -341,13 +341,9 @@ module.exports = registerElement('a-scene', {
...
       value: function () {
-        var canvas = this.canvas;
-        // Set at startup. To enable/disable antialias
-        // at runttime we would have to recreate the whole context
-        var antialias = this.getAttribute('antialias') === 'true';
         var renderer = this.renderer = new THREE.WebGLRenderer({
-          canvas: canvas,
-          antialias: antialias || window.hasNativeWebVRImplementation,
+          canvas: this.canvas,
+          antialias: shouldAntiAlias(this),
```
Comment (cvan) [inline src/core/scene/a-scene.js]: could this instead by exposed as an `antialias` component on `<a-scene>`? that way folks can force enable/disable it if they want.

File: src/core/scene/a-scene.js [hunk 5]
```diff
@@ -518,3 +514,19 @@ function exitFullscreen () {
     document.webkitExitFullscreen();
   }
 }
+
+/**
+ * Determines if renderer anti-aliasing should be enabled.
+ * Enabled by default if has native WebVR or is desktop.
```
Comment (cvan) [inline src/core/scene/a-scene.js]: can this be mentioned in the Docs? (especially if you go the `antialias` component route)

File: src/core/scene/a-scene.js (hunk 4 above)
Comment (dmarcos) [inline src/core/scene/a-scene.js]: antialias cannot be disabled once the context has been created. You would have to create a new one and resubmit all textures and buffers (http://stackoverflow.com/questions/27554969/dynamically-turn-on-off-antialiasing-and-shadows-in-webglrenderer).

Comment (ngokevin) [inline src/core/scene/a-scene.js]: There's a code quality issue to make a `renderer` component to be configurable, `antialias` would be part of that.

Comment (dmarcos) [inline src/core/scene/a-scene.js]: I still don't think it's could be modifiable at runtime. You would invalidate all the components that have submitted textures / buffers to the GPU.

Comment (dmarcos) [inline src/core/scene/a-scene.js]: I pretty elaborated machinery could be implemented to make antialias dynamic but that's something you don't want to do in 99.9999999% of the cases. We can revisit in the future if there's a very good reason

**Comment by dmarcos** [comment]
Thanks!

File: src/core/scene/a-scene.js (hunk 4 above)
Comment (ngokevin) [inline src/core/scene/a-scene.js]: Yeah, I agree, it cannot be configurable at runtime. I just mean configurable at init time.

**Pull request merged**